src/requests/typings/deliveryFunction.ts

```diff
@@ -0,0 +1,65 @@
+// Remove when this eslint rule covers all the cases
+// https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/ROADMAP.md
+/*eslint-disable no-unused-vars*/
+
+export enum DeliveryFunctionRequestEventType {
+  GRAPHQL_FIELD_MAPPING = 'graphql.field.mapping',
+  GRAPHQL_QUERY = 'graphql.query',
+}
+
+export type GraphQLFieldTypeMappingRequest = {
+  fields: { contentTypeId: string; field: Field }[]
+}
+
+type Field = {
+  id: string
+  type: string
+}
+
+export type GraphQLFieldTypeMapping = {
+  contentTypeId: string
+  fieldId: string
+  graphQLOutputType: string
+  graphQLQueryField: string
+  graphQLQueryArgument: string
+}
+
+export type GraphQLFieldTypeMappingResponse = GraphQLFieldTypeMapping[]
+
+export type GraphQLQueryRequest = {
+  query: string
+  variables: Record<string, unknown>
+  operationName?: string
+}
+
+/**
+ * @see https://spec.graphql.org/October2021/#sec-Response
+ */
+export type GraphQLQueryResponse = {
```

Comment:
Is there a situation where it's ok to have all of them empty?

Reply:
While I don't think it is likely this will happen the spec does allow all values to be `null`